yspryn/hw6/reduce_sum.py

```diff
@@ -0,0 +1,17 @@
+from operator import add
+
+
+def reduce(*args, key: callable, default):
+    if not args:
+        return default
+    res = None
+    for i in args:
+        if res is None:
+            res = i
+        else:
+            res = key(res, i)
+    return res
```

Comment:
and what if args is only 1 item?

Reply:
```
def reduce(*args, key: callable, default):
    if not args:
        return default
    
    res, *args = args
    for i in args:
            res = key(res, i)
    return res
```

Reply:
in case of 1 argument is works as expected)
     reduce(3, key=mul, default=1) => 3
     reduce(1, key=add, default=0) => 1